Mathlib/Probability/Moments.lean

```diff
@@ -333,4 +344,363 @@ theorem measure_le_le_exp_cgf [IsFiniteMeasure μ] (ε : ℝ) (ht : t ≤ 0)
+      exact b'
+    have p'' : ∫⁻ ω, ‖rexp (t * X ω)‖₊ ∂μ ≤ ∫⁻ _, ‖rexp (t * b)‖₊ ∂μ := by
+      apply lintegral_mono_ae
+      filter_upwards [p] with ω p
+      simp only [ENNReal.coe_le_coe]
+      rw [← (by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * b)‖₊ = rexp (t * b))] at p
+      exact p
+    suffices ∫⁻ _, ↑‖rexp (t * b)‖₊ ∂μ < ⊤ from lt_of_le_of_lt p'' this
+    simp only [lintegral_const]
+    apply ENNReal.mul_lt_top ENNReal.coe_lt_top (IsFiniteMeasure.measure_univ_lt_top)
+  exact ⟨aestronglyMeasurable_iff_aemeasurable.mpr <|
+    measurable_exp.comp_aemeasurable' (hX.const_mul t), hm1⟩
+
+lemma integrable_expt_bound [IsFiniteMeasure μ] {X : Ω → ℝ} {t a b : ℝ} (hX : AEMeasurable X μ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable (fun ω ↦ exp (t * (X ω))) μ := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  cases lt_trichotomy t 0 with
+  | inr ht => cases ht with
+    | inr ht => exact integrable_expt X t b ht hX (h.mono fun ω h ↦ h.2)
+    | inl ht => rw [ht]; simp only [zero_mul, exp_zero, integrable_const]
+  | inl ht =>
+    rw [(by ext ω; rw [(by ring : - t * (- X ω) = t * X ω)] :
+      (fun ω ↦ rexp (t * X ω)) = (fun ω ↦ rexp (- t * (- X ω))))]
+    exact integrable_expt _ _ _ (by linarith : -t > 0) (AEMeasurable.neg hX)
+      (by filter_upwards [ha] with ω ha using neg_le_neg ha)
+
+lemma tilt_var_bound [IsProbabilityMeasure μ] (a b t : ℝ) (X : Ω → ℝ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b)
+    (hX : AEMeasurable X μ) :
+    variance X (μ.tilted (fun ω ↦ t * X ω)) ≤ ((b - a) / 2) ^ 2 := by
+  have _ : IsProbabilityMeasure (μ.tilted fun ω ↦ t * X ω) :=
+    isProbabilityMeasure_tilted (integrable_expt_bound hX h)
+  exact variance_le_sq_of_bounded
+      ((tilted_absolutelyContinuous μ fun ω ↦ t * X ω) h)
+      (AEMeasurable.mono_ac hX (tilted_absolutelyContinuous μ fun ω ↦ t * X ω))
+
+theorem integrable_bounded [IsFiniteMeasure μ] (a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable X μ := by
+  apply MeasureTheory.memℒp_one_iff_integrable.mp
+  apply memℒp_of_bounded h (aestronglyMeasurable_iff_aemeasurable.mpr hX)
+
+/-- Derivation of `mgf X μ t` is `μ[exp (t * X ω) * X ω]`.
+In order to deal with the differentiation of parametric integrals,
+`hasDerivAt_integral_of_dominated_loc_of_deriv_le` are used in the proof. -/
+theorem tilt_first_deriv [IsFiniteMeasure μ] (t a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b):
+    let g := fun t ↦ mgf X μ t
+    let g' := fun t ↦ μ[fun ω ↦ rexp (t * X ω) * X ω]
+    Integrable (fun ω ↦ rexp (t * X ω) * X ω) μ → HasDerivAt g (g' t) t := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  have hb : ∀ᵐ ω ∂μ, X ω ≤ b := h.mono fun ω h ↦ h.2
+  set c := max ‖a‖ ‖b‖
+  set e := (fun t ↦ fun ω ↦ rexp (t * X ω))
+  set e' := (fun t ↦ fun ω ↦ rexp (t * X ω) * X ω)
+  suffices MeasureTheory.Integrable (e' t) μ ∧
+    HasDerivAt (fun t ↦ μ[e t]) (μ[e' t]) t from by
```

Comment:
If I understand correctly, your goal before that `suffices` has the shape `Integrable ... → HasDerivAt ...` and you replace that implication by `Integrable ... ∧ HasDerivAt ...` for the same functions. That's just making the goal harder, so I don't understand why you do that. Am I missing something?

Reply:
I'm trying to apply `hasDerivAt_integral_of_dominated_loc_of_deriv_le` (Derivative of Integral of Functions with Bounded Derivatives in a Neighborhood), and the conclusion of the statement is `Integrable (F' x₀) ∧ HasDerivAt (fun n => ∫ (a : α), F n a ∂μ) (∫ (a : α), F' x₀ a ∂μ) x₀`, so I had to arrange it in that way.